app/models/event.rb

```diff
@@ -143,4 +144,10 @@ def calc_duration #This is also used in timecards; it should be extracted out
   def trim_id_code
     self.id_code = self.id_code.split[0].downcase
   end
+
+  def expired?
+    Event.where(id_code: id_code)
+      .select { |event| event.end_time > 6.months.ago }.empty?
```

Comment:
Not sure if there's a constraint I'm not aware of, but may I suggest this instead? `Event.where(id_code: id_code).where("end_time > ?", 6.months.ago).empty?`

Chaining the `where` clause allows the database to return yes or no, saving us the work of constructing the objects in ruby, searching through them, only to immediately count and discard them.

Reply:
Also, if the event is already persisted, this search needs to exclude `self.id`, or the record may appear to conflict with itself.